packages/common/src/hooks/useDialog/useDialog.tsx

```diff
@@ -90,9 +90,10 @@ export const useDialog = (dialogProps?: DialogProps): DialogState => {
       // TODO: If you want to change the slide direction or other animation details, add a prop
       // slideAnimationConfig and add a parameter to `useSlideAnimation` to pass in the config.
       WrappedNextButton = React.cloneElement(nextButton, {
-        onClick: () => {
-          onTriggerSlide();
-          onClick();
+        onClick: async () => {
+          const result = await onClick();
+          if (!!result) onTriggerSlide();
+          return result;
```

Comment:
I've just done some adjusting here so that the animation after clicking `onNext` can be stopped and also waits for the `onClick`.

Not the best, I think, as I'd prefer that the animation happen by default and you can return something so that it doesn't happen 🤔